src/test/java/jenkins/plugins/git/GitToolChooserTest.java

```diff
@@ -0,0 +1,478 @@
+    Scenario 1: Size of repo is < 5 MiB, "jgit" should be recommended
+     */
+    @Test
+    public void testGitToolChooserWithLessThan5Mb() throws Exception {
+        String url = "https:github.com";
+        Item context = Mockito.mock(Item.class);
+        String credentialsId = null;
+
+        // With JGit, we don't ask the name and home of the tool
+        GitTool tool = new GitTool("my-git", "/usr/bin/git", Collections.<ToolProperty<?>>emptyList());
+        jenkins.jenkins.getDescriptorByType(GitTool.DescriptorImpl.class).setInstallations(tool);
+
+        GitToolChooser gitToolChooser = new GitToolChooser(url, context, credentialsId, tool.getGitExe());
+
+        //According to size of repo, "jgit" should be recommended but it is not installed by the user
+        //Hence, in this case GitToolChooser resolve gitExe as the user configured `home` value
+        assertThat(gitToolChooser.getGitTool(), is("/usr/bin/git"));
+
+    }
+
+    @Test
+    public void testGitToolChooserWithLessThan5Mb2() throws Exception {
+        String url = "https:github.com";
+        Item context = Mockito.mock(Item.class);
+        String credentialsId = null;
+
+        // With JGit, we don't ask the name and home of the tool
+        GitTool tool = new GitTool("my-git", "/usr/bin/git", Collections.<ToolProperty<?>>emptyList());
+        GitTool jgitTool = new JGitTool(Collections.<ToolProperty<?>>emptyList());
+        jenkins.jenkins.getDescriptorByType(GitTool.DescriptorImpl.class).setInstallations(tool, jgitTool);
+
+        GitToolChooser gitToolChooser = new GitToolChooser(url, context, credentialsId, tool.getGitExe());
+        assertThat(gitToolChooser.getGitTool(), is("jgit"));
+    }
+
+    /*
+    According to the size of repo, GitToolChooser will recommend "jgit" even if "jgitapache" is present
+     */
+    @Test
+    public void testGitToolChooserWithLessThan5Mb3() throws Exception {
+        String url = "https:github.com";
+        Item context = Mockito.mock(Item.class);
+        String credentialsId = null;
+
+        // With JGit, we don't ask the name and home of the tool
+        GitTool tool = new GitTool("my-git", "/usr/bin/git", Collections.<ToolProperty<?>>emptyList());
+        GitTool jgitTool = new JGitTool(Collections.<ToolProperty<?>>emptyList());
+        GitTool jGitApacheTool = new JGitApacheTool(Collections.<ToolProperty<?>>emptyList());
+        jenkins.jenkins.getDescriptorByType(GitTool.DescriptorImpl.class).setInstallations(tool, jgitTool, jGitApacheTool);
+
+        GitToolChooser gitToolChooser = new GitToolChooser(url, context, credentialsId, tool.getGitExe());
+        assertThat(gitToolChooser.getGitTool(), is("jgit"));
+    }
+
+    /*
+    According to the size of repo, GitToolChooser will recommend "jgitapache" since that is user's configured choice
+     */
+    @Test
+    public void testGitToolChooserWithLessThan5Mb4() throws Exception {
+        String url = "https:github.com";
```

Comment:
stop it Justin! 😆 
```suggestion
        String url = "https://github.com";
```

Reply:
@justinharringa haha, sorry for wasting your time on my mistakes!

Reply:
No time wasted my friend!